tests/std/tests/P0881R7_stacktrace/test.cpp

```diff
@@ -0,0 +1,278 @@
+    assert(filesystem::path(three_excluding_top[1].source_file()).filename() == ""sv);
+    assert(filesystem::path(three_excluding_top[2].source_file()).filename() == ""sv);
+
+    assert(three_excluding_top[0].source_line() == 0);
+    assert(three_excluding_top[1].source_line() == 0);
+    assert(three_excluding_top[2].source_line() == 0);
+#endif // ^^^ !HAS_DEBUG_INFO ^^^
+
+#ifdef HAS_NAMES
+    assert(trim_past_plus(three_excluding_top[0].description()) == "P0881R7_stacktrace!three_excluding_top_inner"sv);
+    assert(trim_past_plus(three_excluding_top[1].description()) == "P0881R7_stacktrace!three_excluding_top_outer"sv);
+    assert(
+        trim_past_plus(three_excluding_top[2].description()) == "P0881R7_stacktrace!three_excluding_top_outermost"sv);
+#else // ^^^ HAS_NAMES ^^^ / vvv !HAS_NAMES vvv
+    assert(trim_past_plus(three_excluding_top[0].description()) == "P0881R7_stacktrace"sv);
+    assert(trim_past_plus(three_excluding_top[1].description()) == "P0881R7_stacktrace"sv);
+    assert(trim_past_plus(three_excluding_top[2].description()) == "P0881R7_stacktrace"sv);
+#endif // ^^^ !HAS_NAMES ^^^
+
+    try {
+        (void) all.at(all.size());
+        assert(false); // should have thrown
+    } catch (out_of_range) {
+    }
+
+    auto all_copy = all;
+
+    assert(all == all_copy);
+    assert(all != all_but_top);
+    assert(all > all_but_top);
+    assert(three_excluding_top < all_but_top);
+
+    assert((all <=> all_copy) == strong_ordering::equal);
+    assert((all <=> all_but_top) == strong_ordering::greater);
+    assert((three_excluding_top <=> all_but_top) == strong_ordering::less);
+
+    assert(hash<stacktrace>{}(all) == hash<stacktrace>{}(all_copy));
+    assert(hash<stacktrace_entry>{}(all[0]) == hash<stacktrace_entry>{}(all_copy[0]));
+
+    assert(!all.empty());
+    assert(distance(all.begin(), all.end()) == static_cast<ptrdiff_t>(all.size()));
+    assert(distance(all.rbegin(), all.rend()) == static_cast<ptrdiff_t>(all.size()));
+    assert(distance(all.cbegin(), all.cend()) == static_cast<ptrdiff_t>(all.size()));
+    assert(distance(all.crbegin(), all.crend()) == static_cast<ptrdiff_t>(all.size()));
+
+    stacktrace empty_trace;
+    assert(empty_trace.size() == 0);
+    assert(empty_trace.begin() == empty_trace.end());
+    assert(empty_trace.rbegin() == empty_trace.rend());
+    assert(empty_trace.cbegin() == empty_trace.cend());
+    assert(empty_trace.crbegin() == empty_trace.crend());
+
+    stacktrace_entry empty_entry;
+    assert(empty_entry.description() == ""sv);
+    assert(empty_entry.source_file() == ""sv);
+    assert(empty_entry.source_line() == 0);
+
+    auto s = to_string_using_low_level_members(all);
+    assert(s == to_string_using_stream_entry(all));
+    assert(s == to_string_using_stream(all));
+    assert(s == to_string_using_to_string_entry(all));
+    assert(s == to_string_using_to_string(all));
+}
+
+int main() {
+    std::thread t{test_impl};
+    test_impl();
+    t.join();

```

Comment:
This test is already `using namespace std;`. Additionally, `jthread` would make the `join()` unnecessary.